### DISCUSSION TITLE
directus image not working in kubernetes

### DISCUSSION THREAD

**Discussion by SawantDarshan** [directus/directus#17188] [Archive]
I am trying run **FROM directus/directus:latest** in kubernetes.
After deployment getting following issues - 
`
npm ERR! code EACCES  
npm ERR! syscall mkdir
npm ERR! path /.npm   
npm ERR! errno -13
npm ERR!
npm ERR! Your cache folder contains root-owned files, due to a bug in
npm ERR! previous versions of npm which has since been addressed.
npm ERR!
npm ERR! To permanently fix this problem, please run:
npm ERR!   sudo chown -R 6005:6005 "/.npm"

npm ERR! Log files were not written due to an error writing to the directory: /.npm/_logs
npm ERR! You can rerun the command with `--loglevel=verbose` to see the logs in your terminal
`

**Comment by SaiSuman-N** [comment]
I am also facing same issues. Need support from Directus team

**Reply by rijkvanzanten** [reply]
Unless there's a bug that can be reproduced based on the provided steps, there's nothing we can do to help with hosting specific quirks

**Comment by paescuj** [comment]
Can you share the whole Dockerfile? It seems like there's something wrong in it rather than the `directus` image being a problem.

**Comment by SawantDarshan** [comment]
@rijkvanzanten @paescuj We found the issue and solved. 
So issue was docker build as on kubernetes level we were attaching PV which we are forcing to use security contexts. 
So at the time of building docker it was creating folders with different UID and at the run time security context were overriding this UID.
This are the changes we did inside dockerfile to support security contexts of kubernetes.

<------ Older docker file -----> 
`FROM node:alpine

RUN apk update && apk add python3
ENV PATH="/usr/bin:${PATH}"
WORKDIR /app
COPY . .
RUN npm install -g npm@9.2.0
RUN npm install
RUN npm run build
CMD ["npm", "run", "start"]` 


<------ New docker file -----> 
`FROM node:alpine

RUN apk update && apk add python3
ENV PATH="/usr/bin:${PATH}"
RUN adduser -D -u 6005 contentrpl
WORKDIR /home/node
COPY . .
RUN chown -R contentrpl:contentrpl /usr/local/lib/node_modules
RUN chown -R contentrpl:contentrpl /usr/local/bin
RUN chown -R contentrpl:contentrpl /usr/local/share
RUN chown -R contentrpl:contentrpl /home/node
USER contentrpl 
RUN npm install -g npm@9.2.0 --user=contentrpl
RUN npm install --user=contentrpl
RUN npm run build --user=contentrpl
CMD ["npm", "run", "start"]`

we created user and given them all the access which required at the start of application.
links: 
https://kubernetes.io/docs/tasks/configure-pod-container/security-context/

**Reply by paescuj** [reply]
Thanks for reporting back!

A small note on the side: I know too little about your setup, but maybe it would be possible to just use the [`node` user](https://github.com/directus/directus/blob/8cbaabd8cd98b94973f8715bf8e3e10f455bd745/docker/Dockerfile#L46-L47) from the official image instead of configuring a new one.
